Common/src/toolboxes/inlet_interpolation_functions.cpp

```diff
@@ -0,0 +1,206 @@
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+
+#include "../../include/toolboxes/inlet_interpolation_functions.hpp"
+
+
+CInletInterpolation::CInletInterpolation(CGeometry **geometry, CConfig *config,string profile_filename,unsigned short KIND_MARKER, unsigned short iMarker, unsigned short jMarker, unsigned short nDim){
+
+  nColumns = profileReader.GetNumberOfColumnsInProfile(jMarker);
+  CMarkerProfileReaderFVM profileReader(geometry[MESH_0], config, profile_filename, KIND_MARKER, 52);
+  nRow = profileReader.GetNumberOfRowsInProfile(jMarker);
+  Inlet_Data = profileReader.GetDataForProfile(jMarker);
+  Inlet_Values.resize(nColumns + nDim);
+  InletInterpolatedData.resize((nColumns + nDim)*geometry[MESH_0]->nVertex[iMarker]);
+
+  SetVertex();
+
+  if(config->GetPrintInlet_InterpolatedData() == true)
+    PrintInterpolatedData();
+
+}
+
+CInletInterpolation::~CInletInterpolation(void){}
+
+void CInletInterpolation::Interpolate()
+{
+  switch(config->GetKindInletInterpolationFunction()){
+    case(ONED_LINEAR_SPANWISE):
+      LinearInterpolation();
+    case(ONED_AKIMASPLINE_SPANWISE):
+      AkimaInterpolation(iRow);
+  }
+}
+
+
+void CInletInterpolation::SetVertex(){
+
+  for (iVertex = 0; iVertex < geometry[MESH_0]->nVertex[iMarker]; iVertex++){
+    iPoint   = geometry[MESH_0]->vertex[iMarker][iVertex]->GetNode();
+    Coord    = geometry[MESH_0]->node[iPoint]->GetCoord();  
+
+    Interp_Radius = sqrt(pow(Coord[0],2)+ pow(Coord[1],2));
+    Theta = atan2(Coord[1],Coord[0]);  
+
+    for  (unsigned short iVar=0; iVar < nDim; iVar++)
+      Inlet_Values[iVar]=Coord[iVar];
+
+    for (iRow = 0; iRow < profileReader.GetNumberOfRowsInProfile(jMarker); iRow++) {
+    /*--- Find the two closest radii for the specific vertex. ---*/
+      if (Inlet_Data[nColumns*iRow] <= Interp_Radius && Inlet_Data[nColumns*(iRow+1)] >= Interp_Radius){
+
+      for (index=1; index<nColumns; index++){
+      Point_Match = true;
+
+      Interpolate();
+      
+      if (index > nDim+1)
+        Inlet_Values[index+(nDim-1)+1]=interpolated_value;
+      else
+        Inlet_Values[index+(nDim-1)]=interpolated_value;
+      }
+
+      /*--- New interpolated parameters for that Interp_Radius. ---*/
+      Parameter1=Inlet_Values[nDim+2];
+      Parameter2=Inlet_Values[nDim+3];
+
+      CorrectForInterpolationType();
+
+      Inlet_Values[nDim+2] = unit_r*cos(Theta) - unit_Theta*sin(Theta); //for ix
+      Inlet_Values[nDim+3] = unit_r*sin(Theta) + unit_Theta*cos(Theta); //for iy
+      Inlet_Values[nDim+4] = sqrt(1-pow(unit_r,2)- pow(unit_Theta,2));  //for iz
+
+      SetInterpolatedData();
```

Comment:
I intentionally did that so that I may reduce the number of arguments passing to and fro between functions. It does make more sense to have clear inputs and outputs and I will change my functions accordingly.